sympy/functions/special/delta_functions.py

```diff
@@ -159,6 +159,36 @@ def is_simple(self, x):
             return p.degree() == 1
         return False
 
+    def _eval_rewrite_as_Piecewise(self, *args):
+        """Represents DiracDelta in a Piecewise form
+
+           Examples
+           ========
+
+           >>> from sympy import DiracDelta, Piecewise, Symbol, oo
+           >>> x = Symbol('x', real=True)
+
+           >>> DiracDelta(x).rewrite(Piecewise)
+           Piecewise((oo, Eq(x, 0)), (0, True))
+
+           >>> DiracDelta(x - 5).rewrite(Piecewise)
+           Piecewise((oo, Eq(x - 5, 0)), (0, True))
+
+           >>> DiracDelta(x - 5, 4).rewrite(Piecewise)
```

Comment:
I felt the same way. The documentation is also not very descriptive.

On 17 May 2016 at 08:54, Jason K. Moore <notifications@github.com> wrote:

> In sympy/functions/special/delta_functions.py
> <https://github.com/sympy/sympy/pull/11103#discussion_r63459708>:
>
> > +    def _eval_rewrite_as_Piecewise(self, *args):
> > +        """Represents DiracDelta in a Piecewise form
> > +
> > +           Examples
> > +           ========
> > +
> > +           >>> from sympy import DiracDelta, Piecewise, Symbol, oo
> > +           >>> x = Symbol('x', real=True)
> > +
> > +           >>> DiracDelta(x).rewrite(Piecewise)
> > +           Piecewise((oo, Eq(x, 0)), (0, True))
> > +
> > +           >>> DiracDelta(x - 5).rewrite(Piecewise)
> > +           Piecewise((oo, Eq(x - 5, 0)), (0, True))
> > +
> > +           >>> DiracDelta(x - 5, 4).rewrite(Piecewise)
>
> I don't understand what these "kth derivatice delta" functions are.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/sympy/sympy/pull/11103/files/98beea890a1bf256d01a734f885b21bebf1de9e5#r63459708>
>



-- 
Regards
*Sartaj Singh*

*Mathematics and Computing*,
Indian Institute of Technology,
Varanasi - 221 005 INDIA

E-mail: singhsartaj94@gmail.com, *sartaj.singh.apm13@itbhu.ac.in
<sartaj.singh.apm13@itbhu.ac.in>*